pkg/engine/engine.go

```diff
@@ -190,6 +191,10 @@ func (e *Engine) execAndReplace(args ...string) error {
 
 // imageFQDN builds an image format string from container ref values
 func (c ContainerRef) imageFQDN() string {
+	if c.Image.Name == "" || c.Image.Tag == "" {
+		return ""
+	}
```

Comment:
Had to update the above block because it was _always_ returning `:` if no Image Name or Tag were provided. 

While I understand that's _probably_ okay because they have defaults set during ContainerRef creation as part of the CLI process, this was unexpected from a unit-testing perspective so I just updated this to return an empty string if there's no image references defined.